config/configretry/backoff.go

```diff
@@ -59,7 +59,13 @@ func (bs *BackOffConfig) Validate() error {
 		return errors.New("'max_interval' must be non-negative")
 	}
 	if bs.MaxElapsedTime < 0 {
-		return errors.New("'max_elapsed' time must be non-negative")
+		return errors.New("'max_elapsed_time' must be non-negative")
+	}
+	if bs.MaxElapsedTime < bs.InitialInterval {
+		return errors.New("'max_elapsed_time' must be larger than 'initial_interval'")
+	}
+	if bs.MaxElapsedTime < bs.MaxInterval {
+		return errors.New("'max_elapsed_time' must be larger than 'max_interval'")
```

Comment:
nit: this is technically more correct, maybe not the best wording though
```suggestion
		return errors.New("'max_elapsed_time' must not be less than 'max_interval'")
```